### DISCUSSION TITLE
*.cpp and *.h files - Examples/Projects/RollingClockWithWifiManager/RollingClockWithWifiManager/RollingClockLogic.h

### DISCUSSION THREAD

**Discussion by danpeirce** [witnessmenow/ESP32-Cheap-Yellow-Display#98] [Q&A] [answered]
I should start by saying I am relatively new to Arduino but have decades of experience with C.

**My question is why put function definitions into RollingClockLogic.h** 

I have seen this in other places in Arduino projects but coming from a C background it would seem more appropriate to put the definitions in a *.cpp file and declarations (function prototypes) in a *.h file.

**Comment by mdkendall** [comment, marked as answer]
The pattern is known as a "header-only library". You will not find it difficult to find heated discussions about it, with passionate supporters and detractors. I think there must be something about the Arduino culture or toolchain that encourages this approach, because it is something I have only ever seen in the Arduino ecosystem.

**Reply by danpeirce** [reply]
Thank you for the answer. 
I searched and found no compelling explanation.
